pkg/api/v1/routes/handlers.go

```diff
@@ -79,8 +79,54 @@ func (r *Routes) serverConditionCreate(c *gin.Context) (int, *v1types.ServerResp
 		}
 	}
 
+	return r.conditionCreate(otelCtx, c, kind, serverID)
+}
+
+func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
+	otelCtx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.serverEnroll")
+	span.SetAttributes(
+		attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(c.Param("id"))},
+		attribute.KeyValue{Key: "IP", Value: attribute.StringValue(c.Param("ip"))},
+		attribute.KeyValue{Key: "user", Value: attribute.StringValue(c.Param("user"))})
+	defer span.End()
+
+	serverID, err := uuid.Parse(c.Param("uuid"))
+	if err != nil {
+		r.logger.WithFields(logrus.Fields{
+			"serverID": c.Param("uuid"),
+		}).Info("bad serverID")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Creates a server record in FleetDB.
+	if err = r.repository.AddServer(c.Request.Context(), serverID, c.Param("ip"), c.Param("user"), c.Param("pwd")); err != nil {
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Reuse the inventory for server enrollment.
+	// May change in the future.
+	kind := rctypes.Kind("inventory")
+	if !r.conditionKindValid(kind) {
+		r.logger.WithFields(logrus.Fields{
+			"kind": kind,
+		}).Info("unsupported condition kind")
+
+		return http.StatusBadRequest, &v1types.ServerResponse{
+			Message: "unsupported condition kind: " + string(kind),
+		}
+	}
+
+	return r.conditionCreate(otelCtx, c, kind, serverID)
+}
+
+func (r *Routes) conditionCreate(otelCtx context.Context, c *gin.Context, kind rctypes.Kind, serverID uuid.UUID) (int, *v1types.ServerResponse) {
```

Comment:
I suggest we split this `serverConditionCreate` function, leave the previous name as is and create another function `conditionCreate` with contents from line https://github.com/metal-toolbox/conditionorc/blob/ed108837070cc103684754430d0b8383a72a7ec8/pkg/api/v1/routes/handlers.go#L213

This is because we don't require to perform any validation that the current `serverConditionCreate` method does - since this is new server.

Reply:
Likewise, the `gin.Context` is not going to have a POST payload that fits here. I agree with Joel's comment that we should pull the common `createCondition` elements into a new, un-exported function and call that from `serverEnroll`.

Don't go too crazy updating the existing `conditionCreate`, I will be making a bunch of changes in that area.

Reply:
Thanks for the suggetstion Joel and Vince!
I simply wrap codes starting from line 213 into a function following Joels suggestions in order to bypass the unnecessary validation checks.
PTAL.